src/frontend/src/flows/recovery/useRecovery.ts

```diff
@@ -53,5 +57,75 @@ const runRecovery = async (
     return window.location.reload();
   }
 
+  const deviceAlias = await promptDeviceAlias({
+    title: "Remember this Device",
+    message: "Recovery successful. What device are you using?",
+    cancelText: "Skip",
+  });
+  if (deviceAlias !== null) {
+    // Offer to enroll a new authenticator
+    await enrollAuthenticator({
+      connection: res.connection,
+      userNumber: res.userNumber,
+      deviceAlias,
+    });
+  }
+
   void renderManage(res.userNumber, res.connection);
 };
+
+// Offer to enroll a new device
+const enrollAuthenticator = async ({
+  connection,
+  userNumber,
+  deviceAlias,
+}: {
+  connection: AuthenticatedConnection;
+  userNumber: bigint;
+  deviceAlias: string;
+}): Promise<void> => {
+  let newDevice;
+  try {
+    newDevice = await constructIdentity({
```

Comment:
I would expect to get to try again on error here instead of going to the management screen.

Reply:
Another thing I noticed: (again Firefox) the WebAuthn prompt appears noticably slower then when going through the registration flow (i.e. after pressing "create new"). Do you know why that is?

Reply:
> I would expect to get to try again on error here instead of going to the management screen.

Ah yes, that makes sense 👍 

> Another thing I noticed: on Firefox the WebAuthn prompt appears slower than when going through the registration flow (i.e. after pressing "create new"). Do you know why that is?

I can't think of any reason, but I remember Safari sometimes being very slow for WebAuthn, although a restart did help. Does this happen consistently?

Reply:
@frederikrothenberger actually will need to figure out what exactly we want here. Are you ok with this being done subsequently?

Reply:
> Does this happen consistently?

It happens with all browsers I tested FF, Chrome and Safari, consistently. Maybe it's also just my machine 🤷 

Yeah, this can also be improved later. 👍

Reply:
Ah I see! ok, that's actually normal because we're first fetching the devices to exclude the credentials